新たに書き加えた部分がかなりあるのでとりあえず上げます

yudaw opened the pull request:
#68_社畜るアプリの実装

yudaw committed:
新たに書き加えた部分がかなりあるのでとりあえず上げます
#68_社畜るアプリの実装

noots-val committed:
QuestionsView,クラス answer関数を修正

noots-val commented on syachikuru/views.py:
```diff
@@ -87,11 +44,12 @@ def draw_fig(request):
+        logging.debug(characteristic_type_points)
 
         # 軸の作成と設定
         N = len(characteristic_types)  # 軸数の決定
```
Nの命名の意味は？

noots-val commented on syachikuru/views.py:
```diff
@@ -10,32 +16,89 @@ class QuestionsView(ListView):
+        logging.debug(characteristic_type_points)
+
+        # 軸の作成と設定
+        N = len(characteristic_types)  # 軸数の決定
```
Nの命名の意味は？

yudaw replied:
Nはnumber_of_axに変更

noots-val replied:
axも略さないでほしいかな……。
数文字を節約することによって発生する数秒の思考のラグがもったいない。
flgとかdictくらい一般的ないしは通例的になっているなら話は別だけど……。

noots-val commented on syachikuru/views.py:
```diff
@@ -10,32 +16,89 @@ class QuestionsView(ListView):
+
+        # 軸の作成と設定
+        N = len(characteristic_types)  # 軸数の決定
+        x_as = [2 * pi * n / float(N) for n in range(N)]
```
asとは何かの略ですか？

noots-val replied:
floatを使う理由は？

noots-val replied:
piとは何かの略ですか？

yudaw replied:
x_asはangle_of_axに変更（軸のそれぞれの角度）
floatは意味なかったかも
piはπ（mathの関数）

noots-val replied:
「piはπ（mathの関数）」←コメントしといて

noots-val commented on syachikuru/views.py:
```diff
@@ -10,32 +16,89 @@ class QuestionsView(ListView):
+        x_as += x_as[:1]
+
+        plt.rc('axes', linewidth=0.5, edgecolor="#888888")  # 軸の位置とカラー
+        # ax = plt.figure().add_axes([0.1, 0.1, 0.8, 0.8], polar=True)
```
このコメントは何？

yudaw replied:
「軸の位置やカラーの設定をするしきですよ～」という意味で書きました（いらない？）

noots-val replied:
処理を円滑に理解するためにコメントを書くので、考える手間が発生するコメントはダメです。

noots-val commented on syachikuru/views.py:
```diff
@@ -10,32 +16,89 @@ class QuestionsView(ListView):
+        N = len(characteristic_types)  # 軸数の決定
+        x_as = [2 * pi * n / float(N) for n in range(N)]
+        characteristic_type_points += characteristic_type_points[:1]
+        x_as += x_as[:1]
```
何をしているのか、いまいち見えてきません。
複雑な処理にはコメントを。

yudaw replied:
了解です。
コメントを挿入しときます。

noots-val commented on syachikuru/views.py:
```diff
@@ -10,32 +16,89 @@ class QuestionsView(ListView):
+
+        plt.rc('axes', linewidth=0.5, edgecolor="#888888")  # 軸の位置とカラー
+        # ax = plt.figure().add_axes([0.1, 0.1, 0.8, 0.8], polar=True)
+        fig = plt.figure(1, figsize=(5, 5))
```
pltとは何かの略ですか？

noots-val replied:
figの命名は本当に適切ですか？

yudaw replied:
pltはmatplotのpyplotについて通例的にpltとするそうです
figはほかにも図を載せることからradar_chartとかのほうがいいかね？

noots-val replied:
> pltはmatplotのpyplotについて通例的にpltとするそうです

承知です。

> figはほかにも図を載せることからradar_chartとかのほうがいいかね？

xxxx_flgもしくはyyyy_chartとか、どんなものかが一発で分かる変数名がいいです。

noots-val commented on syachikuru/views.py:
```diff
@@ -10,32 +16,89 @@ class QuestionsView(ListView):
+    def draw_fig(self, request):
+        from math import pi
+        import matplotlib.pyplot as plt
+        from matplotlib.backends.backend_agg import FigureCanvasAgg as FigureCanvas
```
import文はファイルの最初に記述してください。

yudaw replied:
おK

noots-val commented on syachikuru/views.py:
```diff
@@ -10,32 +16,89 @@ class QuestionsView(ListView):
+        response = HttpResponse(content_type='image/png')
+        canvas.print_png(response)
+        return response
+
```
全体通してマジックナンバー（ソースコード中べた書きの数字）が多く見られます。
後から処理を追う人間は、その数字が何を意味するものなのかを知りません。

また、書き換える際も大変です。同じ意味を持つ数字が複数個所で登場する場合もあるのでは？
DRY原則（Don't repeat yourself）やOAOO原則（Once and Only Once）で示されるように、同じ意味を持つ処理は重複して書かないのが原則（あくまで原則）です。

yudaw replied:
それはかなりの反省点
処理が見づらくなり仮置きしていたので少しずつ直します。

yudaw committed:
社畜るviewsを編集(answerとresult部分)
102行目まではマジックナンバーをかなり減らしました
減らそうと思えばanswerのresult_idを決定する部分とかももちろんできますが
式がかなり膨らみそうなのでとりあえず保留してます。
#68_社畜るアプリの実装

yudaw committed:
社畜るviewsを編集しました
マジックナンバーはかなり減ったと思います。
他の部分もできる箇所もあるのですが、式が膨大化すると考えられる部分が多いのでとりあえず保留しています。
#68_社畜るアプリの実装

yudaw committed:
社畜るviewsの編集
先ほど教わった方法で辞書をpoint_dictに受け取っています。
怖いのでとりあえずコミットします
#68

yudaw committed:
先日編集した社畜るのファイル
cookieが日本語対応していないのが原因か？
#68_社畜るアプリの実装

yudaw committed:
レーダーチャート作成用のJSファイルです
かくにんおなしゃす
#68_社畜るアプリの実装

yudaw committed:
varをconstやletにしました
#68_社畜るアプリの実装

noots-val commented on static/syachikuru/js/drawchart.js:
```diff
@@ -1,13 +1,13 @@
-  var result = null;
-  var indexOfName = cookieName + '=';
-  var AllCookie = document.cookie;
+  let result = null;
```
> let result

でオーケーです

noots-val commented on static/syachikuru/js/drawchart.js:
```diff
@@ -0,0 +1,63 @@
+// cookieデータの取得
+function GetCookie(cookieName) {
+  let result = null;
```
> let  result

でオーケーです。

noots-val commented on static/syachikuru/js/drawchart.js:
```diff
@@ -0,0 +1,63 @@
+// cookieデータの取得
+function GetCookie(cookieName) {
```
やっぱりこのメソッド自体よく分からない。

noots-val replied:
あと、lowerCamelCaseにしてください。
先頭小文字、あとは単語区切りが大文字。

noots-val commented on static/syachikuru/js/drawchart.js:
```diff
@@ -0,0 +1,63 @@
+// cookieデータの取得
+function GetCookie(cookieName) {
+  let result = null;
+  const indexOfName = cookieName + '=';
```
> /+ '='

は必須？

noots-val replied:
Nameとは何の？

noots-val commented on static/syachikuru/js/drawchart.js:
```diff
@@ -0,0 +1,63 @@
+function GetCookie(cookieName) {
+  let result = null;
+  const indexOfName = cookieName + '=';
+  let AllCookie = document.cookie;
```
constでいいのでは？

noots-val commented on static/syachikuru/js/drawchart.js:
```diff
@@ -0,0 +1,63 @@
+
+  let position = AllCookie.indexOf(indexOfName);
+  if (position != -1) {
+    const startIndex = position + indexOfName.length;
```
ここ辺りの処理をもう少し現代的に簡潔に書けるのでは？
こちらで検討します。

noots-val commented on static/syachikuru/js/drawchart.js:
```diff
@@ -0,0 +1,63 @@
+}
+
+// レーダーチャートの作成
+const peacockeryType = GetCookie('PEACOCKERY_TYPE');
```
タイプ名までCookieで送る必要はあるのだろうか……。

noots-val commented on static/syachikuru/js/drawchart.js:
```diff
@@ -0,0 +1,63 @@
+const admissibilityPoint = GetCookie('ADMISSIBILITY_POINT');
+const responsibilityPoint = GetCookie('RESPONSIBILITY_POINT');
+const cooperativenessPoint = GetCookie('COOPERATIVENESS_POINT');
+const ctx = document.getElementById('chart').getContext('2d');
```
改行入れてください。

noots-val commented on static/syachikuru/js/drawchart.js:
```diff
@@ -0,0 +1,63 @@
+    }
+  },
+
+
```
一行詰めてください。

noots-val commented on syachikuru/views.py:
```diff
@@ -1,5 +1,14 @@
+# import matplotlib.pyplot as plt
+# from matplotlib.backends.backend_agg import FigureCanvasAgg as FigureCanvas
+
+import logging
```
ログ消してください

noots-val commented on syachikuru/views.py:
```diff
@@ -10,32 +19,94 @@ class QuestionsView(ListView):
+    #     canvas.print_png(response)
+    #     return response
+    #
+    # """
```
コメントアウトら削除で。

noots-val commented on syachikuru/views.py:
```diff
@@ -10,32 +19,94 @@ class QuestionsView(ListView):
+    # response.set_cookie('LOYALTIES_TYPE', characteristic_types[1])
+    # response.set_cookie('ADMISSIBILITY_TYPE', characteristic_types[2])
+    # response.set_cookie('RESPONSIBILITY_TYPE', characteristic_types[3])
+    # response.set_cookie('COOPERATIVENESS_TYPE', characteristic_types[4])
```
コメントアウトは削除て

yudaw committed:
社畜る結果画面に関する部分を更新しました。
valuesは値を変えただけなのであまり気にしないでください
#100_社畜るアプリの実装

yudaw committed:
社畜るの更新です
cssは一応載せときました
#６８_社畜るアプリの実装
